### PR TITLE
Implemented the ISoulbound interface.

### PR DESCRIPTION
**Pull request by 0xfps**
## Implemented the ISoulbound interface and IERC721Extension interface.

Added the ISoulbound interface and IERC721Extension interface.

## Environment

- Compiler version: 0.8.7
- Target EVM version (as per compiler settings): default
- Framework/IDE (e.g. Truffle or Remix): Remix

- Operating system: Windows

## ISoulbound interface and IERC721Extension interface.

In response to issue #2 and #4, the ISoulbound and IERC721Extension interfaces have been added to the codebase.

### CODE REVIEW

**Comment by 0xfps** [comment]
Also added interface for issue #4.

File: src/contracts/interfaces/ISoulbound.sol [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+* There are some quick information to note when implementing this interface.
+* In addition to the link above, this interface ensures that a token can only be minted to an address once, minting the same token or...
+* ...transferring the same token to another address is not feasible. This is to make sure that, that particular token stays with the address,...
+* ...and CANNOT be moved around.
+* The address the token is minted to is called the 'Soul Address'.
+* However, this token can be revoked by the minter (whoever minted it to the Soul Address), and then, and only then...
+* ...can that token be reminted to the Soul Address.
+* Also, in cases where the token is lost, the above will also apply.
+*/
+
```
Comment (koolamusic) [inline src/contracts/interfaces/ISoulbound.sol]: Add the draft `EIP` we derived this from at the top

File: src/contracts/interfaces/ISoulbound.sol [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    * [CONDITIONS]
+    * `_tokenId` must be an existent token.
+    *
+    * @param:
+    * uint256 _tokenId => Token minted.
+    *
+    * @return:
+    * address.
+    */
+    function ownerOfToken(uint256 _tokenId) external returns(address);
```
Comment (koolamusic) [inline src/contracts/interfaces/ISoulbound.sol]: We should strive to maintain compatibility with the ERC721 spec. So I think we could keep this as `ownerOf`

**Review by koolamusic — CHANGES_REQUESTED** [review]
Looks  good to me. 
Taking a look at the `isMinted` function I think we can have more use for it by leveraging it to write a `balanceOf` function that can return `0 or 1` where

- isMinted evaluates to `true` will return 1
- isMinted evals to `false` will return 0

This maintains compatibility with wallets and indexers but relies on the mechanisms we've defined in soulbound tokens

File: src/contracts/interfaces/ISoulbound.sol [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    * `_tokenId` must be an existent token.
+    *
+    * @param:
+    * address _from => Address which owns token `_tokenId`.
+    * uint256 _tokenId => Token to revoke.
+    *
+    * @return:
+    * bool.
+    */
+    function revoke(address _from, uint256 _tokenId) external returns(bool);
```
Comment (koolamusic) [inline src/contracts/interfaces/ISoulbound.sol:116]: This revoke should only happen by `issuer` so something like a modifier for `onlyIssuer` would apply in this case

**Comment by 0xfps** [comment]
I have made necessary changes.

**Comment by 0xfps** [comment]
Also, added the interface for the POAP smart contract for issue #1.

**Review by koolamusic — APPROVED** [review]

**Pull request merged**